src/transformers/configuration_utils.py

```diff
@@ -667,7 +667,43 @@ def update(self, config_dict: Dict[str, Any]):
         Updates attributes of this class with attributes from ``config_dict``.
 
         Args:
-            config_dict (:obj:`Dict[str, Any]`): Dictionary of attributes that shall be updated for this class.
+            config_dict (:obj:`Dict[str, Any]`): Dictionary of attributes that should be updated for this class.
         """
         for key, value in config_dict.items():
             setattr(self, key, value)
+
+    def update_from_string(self, update_str: str):
+        """
+        Updates attributes of this class with attributes from ``update_str``.
+
+        The expected format is in the form of this example ``key1=10,key2=0.2,key3=true,key4=string`` - i.e. just write
+        ints and floats and strings as is, and for booleans use ``true`` or ``false``.
```

Comment:
If you like it let's discuss if we want perhaps `;` instead of `,` or anything else before we release it.

Should we expect any `,` in the values of the config object?

And at the moment it only support top-level key updates. so no nested dicts.

Reply:
Some config have lists values sometimes (though it's pretty rare), but then we can imagine parsing on space, so , is fine with me. I think we're good with just the focus on int/float/bool/str.